materialize/lifecycle.go

```diff
@@ -520,91 +642,182 @@ func RunTransactions(
 	stream Driver_TransactionsServer,
 	transactor Transactor,
 	log *logrus.Entry,
-) (err error) {
+) (_err error) {
```

Comment:
`RunTransactions` is...rather complex. :slightly_smiling_face:  Would it be helpful to try and get some tests around it? I'm not entirely confident that I'd catch potentials bugs by just reading the source.

Reply:
Yes! I've worked up a TxnClient for driving a Transactor, which will also live in this package, and has abstracted joints for the various bits where we wire it into the Flow runtime. I've also implemented a test of their integration. While not comprehensive, it's better than where we are right now. Commits forthcoming.

(FWIW, there's also a lower-level test of RunTransactions with the SQLite Transactor implementation in the `flow` repo.)